Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -817,25 +817,23 @@ public struct RegularExpression: BSONValue, Equatable, Codable {
         return self.init(pattern: patternString, options: optionsString)
     }
 
-    /// Creates an `NSRegularExpression` with the pattern and options of this `RegularExpression`.
-    /// Note: `NSRegularExpression` does not support the `l` locale dependence option, so it will
-    /// be omitted if set on this `RegularExpression`.
-    public var nsRegularExpression: NSRegularExpression {
-        let opts = NSRegularExpression.optionsFromString(self.options)
-        do {
-            return try NSRegularExpression(pattern: self.pattern, options: opts)
-        } catch {
-            fatalError("Failed to initialize NSRegularExpression with " +
-                "pattern '\(self.pattern)'' and options '\(self.options)'")
-        }
-    }
-
     /// Returns `true` if the two `RegularExpression`s have matching patterns and options, and `false` otherwise.
     public static func == (lhs: RegularExpression, rhs: RegularExpression) -> Bool {
         return lhs.pattern == rhs.pattern && lhs.options == rhs.options
     }
 }
 
+/// Extension of `NSRegularExpression` allowing it to be initialized from a `RegularExpression` `BSONValue`.
+extension NSRegularExpression {
```

Comment:
I think we could probably combine this with the other `NSRegularExpression` extension we have defined earlier in the file.

Reply:
Yep, didn't see that extension the first time. Moved.